public/app/features/playlist/PlaylistSrv.ts

```diff
@@ -78,17 +82,50 @@ export class PlaylistSrv {
     // setup location tracking
     this.locationListenerUnsub = locationService.getHistory().listen(this.locationUpdated);
 
-    return getBackendSrv()
-      .get(`/api/playlists/${playlistUid}`)
-      .then((playlist: any) => {
-        return getBackendSrv()
-          .get(`/api/playlists/${playlistUid}/dashboards`)
-          .then((dashboards: any) => {
-            this.dashboards = dashboards;
-            this.interval = rangeUtil.intervalToMs(playlist.interval);
-            this.next();
-          });
-      });
+    const searcher = getGrafanaSearcher();
+    let urls: string[] = [];
+    const playlist = await getBackendSrv().get<Playlist>(`/api/playlists/${playlistUid}`);
+    if (!playlist.items?.length) {
+      // alert
+      return;
+    }
+    this.interval = rangeUtil.intervalToMs(playlist.interval);
+    for (const item of playlist.items) {
+      if (!item.value) {
+        continue; // invalid item
+      }
+      const query: SearchQuery = { kind: ['dashboard'] };
+      switch (item.type) {
+        case 'dashboard_by_uid':
+          query.uid = [item.value];
+          break;
+
+        case 'dashboard_by_id':
+          query.uid = await getBackendSrv().get<string[]>(`/api/dashboards/ids/${item.value}`);
+          if (!query.uid.length) {
+            continue;
+          }
+          break;
+
+        case 'dashboard_by_tag':
+          query.tags = [item.value];
+          break;
+      }
+
+      const rsp = await searcher.search(query);
```

Comment:
Is this going to result in a call to the backend per item?

Reply:
yes.  🤔 given how the new search is implemented, we could actually do it all in one.

Reply:
I updated this to fetch everything in one request.